nixos/modules/services/web-apps/homebox.nix

```
@@ -43,16 +52,37 @@ in
       group = "homebox";
     };
     users.groups.homebox = { };
-    services.homebox.settings = {
-      HBOX_STORAGE_DATA = mkDefault "/var/lib/homebox/data";
-      HBOX_DATABASE_DRIVER = mkDefault "sqlite3";
-      HBOX_DATABASE_SQLITE_PATH = mkDefault "/var/lib/homebox/data/homebox.db?_pragma=busy_timeout=999&_pragma=journal_mode=WAL&_fk=1";
-      HBOX_OPTIONS_ALLOW_REGISTRATION = mkDefault "false";
-      HBOX_OPTIONS_CHECK_GITHUB_RELEASE = mkDefault "false";
-      HBOX_MODE = mkDefault "production";
+    services.homebox.settings = lib.mkMerge [
+      (lib.mapAttrs (_: mkDefault) {
+        HBOX_STORAGE_DATA = "/var/lib/homebox/data";
+        HBOX_DATABASE_DRIVER = "sqlite3";
+        HBOX_DATABASE_SQLITE_PATH = "/var/lib/homebox/data/homebox.db?_pragma=busy_timeout=999&_pragma=journal_mode=WAL&_fk=1";
+        HBOX_OPTIONS_ALLOW_REGISTRATION = "false";
+        HBOX_OPTIONS_CHECK_GITHUB_RELEASE = "false";
+        HBOX_MODE = "production";
+      })
+
+      (lib.mkIf cfg.database.createLocally {
+        HBOX_DATABASE_DRIVER = "postgres";
+        HBOX_DATABASE_HOST = "/run/postgresql";
+        HBOX_DATABASE_USERNAME = "homebox";
+        HBOX_DATABASE_DATABASE = "homebox";
+        HBOX_DATABASE_PORT = "5432";
```

Urgh just noticed this should use the setting value.

```suggestion
        HBOX_DATABASE_PORT = toString config.services.postgresql.settings.port;
```